model/src/main/java/org/projectnessie/model/Branch.java

```diff
@@ -31,7 +32,7 @@
 @JsonSerialize(as = ImmutableBranch.class)
 @JsonDeserialize(as = ImmutableBranch.class)
 @JsonTypeName("BRANCH")
-public interface Branch extends Reference {
+public interface Branch extends Reference, Serializable {
```

Comment:
we need to keep the `Serializable` on this class because Kryo for some reason requires that the class uses as a key in the map implements `Serializable`. 
The kryo is used for all `ImmutableBranch` read/writes when running spark. I've verified it by validating that those logs:
[ImmutableBranchSerializer#read](https://github.com/projectnessie/nessie/pull/3599/files#diff-19a139a5d1fa76b7773b67373a4f3cca5002f712d4b13e471742d5d99b7db933R44) 
a
https://github.com/projectnessie/nessie/pull/3599/files#diff-19a139a5d1fa76b7773b67373a4f3cca5002f712d4b13e471742d5d99b7db933R36
are executed for each object

Reply:
The `Spark` end-to-end test: `TestJerseyRestGcInMemory`